nilearn/surface/surface.py

```diff
@@ -1975,3 +1992,139 @@ def new_img_like(ref_img, data):
+        Cluster size threshold, in voxels. In the returned thresholded map,
+        sets of connected voxels (``clusters``) with size smaller
+        than this number will be removed.
+
+        Not implemented
+
+    two_sided : :obj:`bool`, default=True
+        Whether the thresholding should yield both positive and negative
+        part of the maps.
+
+    mask_img : Niimg-like object, default=None
+        Mask image applied to mask the input data.
+        If None, no masking will be applied.
+
+    copy : :obj:`bool`, default=True
+        If True, input array is not modified. True by default: the filtering
+        is not performed in-place.
+
+    """
+    img_data = img.data
+
+    if mask_img is not None:
+        check_same_n_vertices(mask_img.mesh, img.mesh)
+        for hemi in mask_img.data.parts:
+            mask_data = mask_img.data.parts[hemi]
+            # Set as 0 for the values which are outside of the mask
+            img_data.parts[hemi][mask_data == 0.0] = 0.0
+
+    cutoff_threshold = check_threshold(
+        threshold,
+        img_data,
+        percentile_func=scoreatpercentile,
+        name="threshold",
+    )
+
+    # Apply threshold
+    if two_sided:
+        img_data[np.abs(img_data) < cutoff_threshold] = 0.0
+    else:
+        img_data[img_data < cutoff_threshold] = 0.0
+
+
+    # Perform cluster thresholding, if requested
+    if cluster_threshold > 0:
+        raise NotImplementedError
+    
+    # Reconstitute img object
+    thresholded_img = new_img_like(img, img_data)
+
+    return thresholded_img
+
+
+def load_mask_img(mask_img, allow_empty=False):
+    """Check that a mask is valid, ie with two values including 0 and load it.
+
+    Parameters
+    ----------
+    mask_img : SurfaceImage object
+        See :ref:`extracting_data`.
+        The mask to check.
+
+    allow_empty : :obj:`bool`, default=False
+        Allow loading an empty mask (full of 0 values).
+
+    Returns
+    -------
+    mask : SurfaceImage
+        Boolean version of the mask.
+    """
+    mask = get_data(mask_img, ensure_finite=True)
+    values = np.unique(mask)
+
+    if len(values) == 1:
+        # We accept a single value if it is not 0 (full true mask).
+        if values[0] == 0 and not allow_empty:
+            raise ValueError(
+                "The mask is invalid as it is empty: it masks all data."
+            )
+    elif len(values) == 2:
+        # If there are 2 different values, one of them must be 0 (background)
+        if 0 not in values:
+            raise ValueError(
+                "Background of the mask must be represented with 0. "
+                f"Given mask contains: {values}."
+            )
+    else:
+        # If there are more than 2 values, the mask is invalid
+        raise ValueError(
+            f"Given mask is not made of 2 values: {values}. "
+            "Cannot interpret as true or false."
+        )
+
+    mask = as_ndarray(mask, dtype=bool)
+    return mask, mask_img.affine
```

Comment:
```suggestion
def load_mask_img(mask_img, allow_empty=False):
    """Check that a mask is valid, ie with two values including 0 and load it.

    Parameters
    ----------
    mask_img : SurfaceImage object
        See :ref:`extracting_data`.
        The mask to check.

    allow_empty : :obj:`bool`, default=False
        Allow loading an empty mask (full of 0 values).

    Returns
    -------
    mask : SurfaceImage
        Boolean version of the mask.
    """
    mask = get_data(mask_img, ensure_finite=True)
    values = np.unique(mask)

    if len(values) == 1:
        # We accept a single value if it is not 0 (full true mask).
        if values[0] == 0 and not allow_empty:
            raise ValueError(
                "The mask is invalid as it is empty: it masks all data."
            )
    elif len(values) == 2:
        # If there are 2 different values, one of them must be 0 (background)
        if 0 not in values:
            raise ValueError(
                "Background of the mask must be represented with 0. "
                f"Given mask contains: {values}."
            )
    else:
        # If there are more than 2 values, the mask is invalid
        raise ValueError(
            f"Given mask is not made of 2 values: {values}. "
            "Cannot interpret as true or false."
        )

    mask = as_ndarray(mask, dtype=bool)
    return mask, mask_img.affine
```


Almost all of this is duplicated from

https://github.com/nilearn/nilearn/blob/cd2871895581caacfcbd2bb443fa5d54304d94b5/nilearn/masking.py#L43-L88

Reply:
consider refactoring